app/code/Magento/GiftMessageGraphQl/Model/Resolver/Cart/GiftMessage.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\GiftMessageGraphQl\Model\Resolver\Cart;
+
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\GraphQl\Config\Element\Field;
+use Magento\Framework\GraphQl\Exception\GraphQlInputException;
+use Magento\Framework\GraphQl\Query\Resolver\ContextInterface;
+use Magento\Framework\GraphQl\Query\Resolver\Value;
+use Magento\Framework\GraphQl\Query\ResolverInterface;
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+use Magento\GiftMessage\Api\CartRepositoryInterface;
+
+/**
+ * Class provides ability to get GiftMessage for cart
+ */
+class GiftMessage implements ResolverInterface
+{
+    /**
+     * @var CartRepositoryInterface
+     */
+    private $cartRepository;
+
+    /**
+     * GiftMessage constructor.
+     *
+     * @param CartRepositoryInterface $cartRepository
+     */
+    public function __construct(
+        CartRepositoryInterface $cartRepository
+    ) {
+        $this->cartRepository = $cartRepository;
+    }
+
+    /**
+     * Return information about Gift message of cart
+     *
+     * @param Field            $field
+     * @param ContextInterface $context
+     * @param ResolveInfo      $info
+     * @param array|null       $value
+     * @param array|null       $args
+     *
+     * @return array|Value|mixed
+     *
+     * @throws GraphQlInputException
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
+        if (!isset($value['model'])) {
+            throw new GraphQlInputException(__('"model" value should be specified'));
+        }
+        $cart = $value['model'];
+
+        $giftCartMessage = $this->cartRepository->get($cart->getId());
```

Comment:
It might not be obvious, but there's a chance to get the exception in case if the cart ID is not correct

https://github.com/magento/magento2/blob/84daa3ff9fcecb49eefe8542c528e2f0ddf9fdec/app/code/Magento/Quote/Model/QuoteRepository.php#L164

So we need to handle this exception correspondingly

Reply:
**Done**